firefox_puppeteer/ui/about_window.py

```diff
@@ -0,0 +1,130 @@
+        check_button = self.marionette.find_element(By.ID, 'checkForUpdatesButton')
+        check_button.click()
+
+        # Wait for the update checking to finish
+        Wait(self.marionette, self.TIMEOUT_UPDATE_CHECK).until(
+            lambda _: self.wizard_state != 'checkingForUpdates')
+
+    def download(self, wait_for_finish=True, timeout=TIMEOUT_UPDATE_DOWNLOAD):
+        """ Download the update
+
+        :param wait_for_finish: Should the function wait until the download has finished?
+        :param timeout: How long to wait for the download to finish
+        """
+        assert self.software_update.update_channel.default_channel ==\
+               self.software_update.update_channel.channel,\
+            'The update channel has been set correctly.'
+
+        if self.wizard_state == 'downloadAndInstall':
+            install_button = self.marionette.find_element(By.ID, 'downloadAndInstallButton')
+            install_button.click()
+
+        # Wait for the download to start
+        Wait(self.marionette).until(
+            lambda _: self.wizard_state != 'downloadAndInstall')
+
+        # If there are incompatible addons we fallback on old software update dialog for updating
+        if self.wizard_state == 'applyBillboard':
+            update_button = self.marionette.find_element(By.ID, 'updateButton')
+            update_button.click()
+
+            # The rest of the code inside this `if` uses the update wizard which is
+            # not being converted yet
+            #
+            # The current JS code is:
+            #
+            #     var wizard = updateWizard.handleUpdateWizardDialog();
+            #     wizard.waitForWizardPage(updateWizard.WIZARD_PAGES.updatesfoundbasic);
+            #     wizard.download();
+            #     wizard.close();
+            #     this._downloadDuration = wizard._downloadDuration;
+            #
+            #     return;
+
+        if wait_for_finish:
+            start_time = datetime.now()
+            self._wait_for_download_finished(timeout)
+            self.download_duration = datetime.now() - start_time
+
+    def _wait_for_download_finished(self, timeout):
+        """ Waits until download is completed
+
+        :param timeout: How long to wait for the download to complete
+        """
+
+        Wait(self.marionette, timeout).until(
+            lambda _: self.wizard_state != 'downloading',
+            message='Download has been completed.')
+
+        assert self.wizard_state != 'downloadFailed'
+
+    def wait_for_update_applied(self, timeout=TIMEOUT_UPDATE_APPLY):
+        """ Waits until the downloaded update has been applied
+
+        :param timeout: How long to wait for the update to apply
+        """
+
+        Wait(self.marionette, timeout).until(
+            lambda _: self.wizard_state == 'apply',
+            message='Final wizard page has been selected.')
+
+        Wait(self.marionette, timeout).until(
```

Comment:
Please leave and improve the comment we had for this code in the Mozmill version of the about window.